### ISSUE TITLE
Demos are broken in major, I think due to a stealjs bug

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/canjs#4225]
If you checkout `major` and open up:

http://localhost:8080/canjs/demos/technology-overview/simple-todos.html

It doesn't load and provides the following error:

```
steal.js:8137 TypeError: Cannot read property 'extend' of undefined
```

This is because `DefineMap` isn't defined here:

```
<script id="demo-source" main="@empty" src="../../node_modules/steal/steal.js" deps-bundle>
import { DefineList, DefineMap, stache, stacheBindings } from "can";

var TodosApp = DefineMap.extend({
...
```

There are a few things I noticed:

1.  Despite having `deps-bundle`, it's not actually trying to load the `deps-bundle`.  There's no request for it.
2.  es/can-define gets transpiled to:
    ```js
    (function() { var __moduleName = "can@5.0.0-pre.2#es/can-define"; var __moduleAddress = "http://localhost:8080/canjs/es/can-define.js";System.register([], function (_export, _context) {
      "use strict";

      return {
       setters: [],
       execute: function () {
         _export("__esModule", true);
        }
      };
    });
    //# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsImZpbGUiOiJjYW4tZGVmaW5lLmpzIiwic291cmNlc0NvbnRlbnQiOltdfQ==
     }).call(__global);
   ```
   This doesn't look like it's exporting any of those modules.  
2. If I change `can.js` from: 
    ```js
    export { define, DefineMap, DefineList } from "./es/can-define";
    ```
    to
    ```js
    export { default as define } from "can-define";
    export { default as DefineMap } from "can-define/map/map";
    export { default as DefineList } from "can-define/list/list";
    ```
     Then `DefineMap` is exported, but `deps-bundle` still isn't used.

So it seems like there might be 3 problems:

1.  Perhaps steal is being too aggressive with "exporter" modules that import other exporter modules (I don't know what we are calling them).  
2. deps-bundle isn't working.
3. Steal doesn't provide nice errors when you use the inline script tags.

For #1, if it's hard to fix, we could have separate mains for webpack and steal.  This would avoid creating 2 extra hops to load something like `can-component` w/i steal.

**Comment by matthewp** [comment]
My guess is that deps bundle probably isn't the cause, I bet the tree-shaking is not able to handle the multiple levels of re-exporting that I added.

**Comment by matthewp** [comment]
Yeah, so it happens without the deps-bundle. Tree-shaking almost definition the problem here, still investigating.

**Comment by matthewp** [comment]
Turns out major was pinned to steal `pre.1`. When I updated it to latest the demos work now (with deps bundle being used).  https://github.com/canjs/canjs/pull/4226

I went ahead and made a test case for the new structure that will be in place once #4224 is merged. So I'll have that in case something breaks there.

**Comment by matthewp** [comment]
Fixed by #4226
